src/Data/ProgrammesDb/EntityRepository/CoreEntityRepository.php

```diff
@@ -196,6 +196,30 @@ public function findByPidFull(string $pid, string $entityType = 'CoreEntity'): ?
         return $this->resolveCategories($withHydratedParents)[0];
     }
 
+    /**
+     * @param string[] $pids
+     * @param string $entityType
+     * @return array
+     */
+    public function findByPids(array $pids, string $entityType = 'CoreEntity'): array
```

Comment:
Seeing as it is so similar, is it worth refactoring this and `findByPid` to use as much of the same code as possible. Instead of duplicating?

Reply:
TBH I deliberately avoided that. findByPid is about the most called function in the codebase, and it actually has some subtly different behaviour that I don't want people breaking if they touch this function. I'll eat the overhead of the duplication to avoid people breaking findByPid